src/main/java/seedu/address/storage/JsonAdaptedPerson.java

```diff
@@ -271,18 +301,28 @@ public Person toModelType() throws IllegalValueException {
         final ShiftDays modelShiftDays = generateShiftDays(shiftDays);
         final Set<Specialisation> modelSpecialisations = generateSpecialisations(specialisations);
 
+
         final List<Affiliation> personAffiliations = generateAffiliationList();
         final List<Affiliation> personAffiliationHistory = generateAffiliationHistoryList();
         final Set<Affiliation> modelAffiliations = new HashSet<>(personAffiliations);
         final Set<Affiliation> modelAffiliationHistory = new HashSet<>(personAffiliationHistory);
 
         Person generatedPerson = modelRole.generatePerson(modelName,
                 modelPhone, modelEmail, modelAffiliations, modelAffiliationHistory);
+
+        if (generatedPerson instanceof Patient) {
+            final NextOfKin modelNextOfKin = generateNextOfKin(nextOfKin);
+            Patient patient = ((Patient) generatedPerson);
+            patient.setNextOfKin(modelNextOfKin);
+            return patient;
```

Comment:
Uses NextOfKin as unique attribute of Patient, good.

Reply:
Thanks! :)